### DISCUSSION TITLE
Crazyfile 2.1 with Bigquad deck, One motor dosen't spin

### DISCUSSION THREAD

**Discussion by Colin-Wang-design** [bitcraze/discussions#1563] [Q&A] [answered]
Four motors were well connected with Bigquad deck using ESCs, but only three motor spinng. I changed that "borken" motor but problem remians the same as before, then I changed a new ESC to that new motor, finaly it was spinned, however, the other motor dosen't work again. 
So, it's always three motors spinning!

Please help me!!

**Comment by tobbeanton** [comment]
What type of motors and ESC do you have? I'm thinking if there could be some signal protocol or signal level issue.

**Reply by Colin-Wang-design** [reply]
Hey,

ESCs are Flycolor X-Cross 35A BLheli_32 and motors are Turnigy D2826-1100KV 265W.

Here is the spining three motors: https://youtube.com/shorts/LWR-4IeYh4g?feature=share

Before connect to Bigquad deck, I tested all four motors by directly connecting ESCs to a additional receiver and all four motors spining smoothly, therefore, I think it's the signal problem from crazyfile for that 'broken' motor.

Thanks for relying!

**Comment by tobbeanton** [comment]
I see no reason why the ESC would not be compatible. Next step would be to measure the motor signal, preferably with a scope to see that you get a signal.

What FW are you running on the crazyflie?

**Reply by Colin-Wang-design** [reply]
Firmware is the latest version, git colned from Github. 

I'm guessing the M3 port was not output signals, and I tested the M3 motor with a receiver, it works fine. I also tried swaped M2 and M3 signal wires, and the motor doesn't sipn changed to M2. 

Here is the test result: https://youtube.com/shorts/EOJbXsPCHO8?si=mC-8iLlGrs915awx
and flying with cfclient: https://youtube.com/shorts/JqabZRZTn7U?feature=shared



Is that possible some points between Bigquad and crazyfile are not well connected, then no output from M3 port?

**Comment by tobbeanton** [comment]
I tried the latest github version and it works fine so it is starting to point towards a hardware problem. Has M3 on the BigQuad deck ever worked? It could be the STM32 GPIO that has broken but more likely that there is a glitch so the signal doesn't make it all the way to the ESC.

Try this.
* Build the firmware with "standard PWM" as motor signal protocol. Flash and start bigquad.
![image](https://github.com/user-attachments/assets/15332520-7e40-4393-8cce-294c0f746111)
*  Measure the voltage on IO_2 with a multi-meter. It should say ~ 2V
![image](https://github.com/user-attachments/assets/7f664b46-1919-4c77-8a16-a048838956ac)

**Reply by Colin-Wang-design** [reply]
M3 was worked once at the begining, then never worked. The voltage on IO_2 is 1.89v.
![_cgi-bin_mmwebwx](https://github.com/user-attachments/assets/a2c8ec68-5ae6-41f0-af8a-c06dac87663c)

**Comment by Colin-Wang-design** [comment, marked as answer]
Hey Tobias,

Thanks a lot for the reply. I changed a new Crazyfile and all motors worked! It seems somewhere has poor connection in that Crazyfile.

**Reply by tobbeanton** [reply]
Great that you got it to work. The 1.89V tells me there is a signal, so like you said, most likely a poor connection somewhere.
